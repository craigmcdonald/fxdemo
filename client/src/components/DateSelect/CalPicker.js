import React from 'react'
import PropTypes from 'prop-types'
import DatePicker from 'react-datepicker/lib/datepicker'
import moment from 'moment'
import style from './CalPicker.scss'
import 'react-datepicker/dist/react-datepicker-cssmodules.css'
const { func, object } = PropTypes

const CalPicker = React.createClass({
  propTypes: {
    minDate: object,
    maxDate: object,
    selected: object,
    onSelect: func
  },
  render() {
    return (
      <div styleName='style.cal-picker'>
        <DatePicker
          minDate={moment().subtract(90,'days')}
          maxDate={moment()}
          selected={this.getValue()}
          onSelect={(val => this.setValue(val))}
        />
      </div>
    )
  },
  getValue() {
    return this.props.selected
  },
  setValue(val) {
    this.props.onSelect(val)
  }
})

export default CalPicker
