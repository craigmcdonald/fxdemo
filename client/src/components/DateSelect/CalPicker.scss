.cal-picker {
  width:250px;
  margin-bottom: 15px;
}
.cal-picker input {
  padding-left:10px;
  font-size: 16px;
  font-family: inherit;
  height: 33px;
  width: 238px;
  border: 1px solid #ccc;
  border-radius: 5px;
}
