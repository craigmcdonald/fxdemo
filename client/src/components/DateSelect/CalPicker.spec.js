import React from 'react'
import ReactDOM from 'react-dom'
import CalPicker from './CalPicker'
import moment from 'moment'
import { mount, shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'


test('CalPicker Snapshot Test', () => {
  let dateVal = new Date("2017-08-09T12:53:14.878Z")
  let minDate = new Date("2017-05-11T12:53:14.876Z")
  function changeDate(val) {
    dateVal = val
  }
  const component = shallow(
    <CalPicker
      selected={dateVal}
      onSelect={changeDate}
      minDate={minDate}
      maxDate={dateVal}
    />
  )
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('CalPicker has an initial date', () => {
  let dateVal = moment()
  let minDate = moment().subtract(90,'days')
  function changeDate(val) {
    dateVal = val
  }
  const dateselect = mount(
    <CalPicker
      selected={dateVal}
      onSelect={changeDate}
      minDate={minDate}
      maxDate={dateVal}
    />
  )
  expect(dateselect.html()).toMatch(moment().format('MM[/]DD[/]YYYY'))
})
