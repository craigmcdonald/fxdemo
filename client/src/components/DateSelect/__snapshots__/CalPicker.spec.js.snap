// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalPicker Snapshot Test 1`] = `
<div
  styleName="style.cal-picker"
>
  <DatePicker
    allowSameDay={false}
    dateFormat="L"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={false}
    dropdownMode="scroll"
    maxDate={"2017-08-09T13:58:06.096Z"}
    minDate={"2017-05-11T13:58:06.095Z"}
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    selected={2017-08-09T12:53:14.878Z}
    utcOffset={60}
    withPortal={false}
  />
</div>
`;
